File: Cpp/gmock_version/textconverter/HtmlTextConverter.cpp

#include "HtmlTextConverter.hpp"
#include "StreamConverter.hpp"
#include "StringEscapeUtils.hpp"

#include <fstream>

HtmlTextConverter::HtmlTextConverter(std::string& filename) :
    m_filename(filename)
{}

std::string HtmlTextConverter::convertToHtml()
{
    std::ifstream reader(m_filename);
    StreamConverter converter(reader);
    return converter.convertToHtml();
}

std::string HtmlTextConverter::getFilename()
{
    return m_filename;
}

File: kotlin/TextConverter/src/main/kotlin/tddmicroexercises/textconvertor/HtmlTextConverter.kt
package tddmicroexercises.textconvertor

import java.io.BufferedReader
import java.io.FileReader
import java.io.IOException

class HtmlTextConverter(val filename: String) {

    @Throws(IOException::class)
    fun convertToHtml(): String {

        val reader = BufferedReader(FileReader(filename))

        var line: String? = reader.readLine()
        var html = ""
        while (line != null) {
            html += StringEscapeUtils.escapeHtml(line)
            html += "<br />"
            line = reader.readLine()
        }
        return html

    }
}
